src/target/devo7e-256/channels.c

```diff
@@ -104,38 +135,100 @@ s32 CHAN_ReadRawInput(int channel)
         case INP_SWB0:     value = global_extra_switches   & 0x01;  break;
         case INP_SWB1:     value = !(global_extra_switches & 0x03); break;
         case INP_SWB2:     value = global_extra_switches   & 0x02;  break;
-        case INP_SWE0:     value = global_extra_switches   & 0x04;  break;
-        case INP_SWE1:     value = !(global_extra_switches & 0x0c); break;
-        case INP_SWF0:     value = global_extra_switches   & 0x01;  break;
-        case INP_SWF1:     value = !(global_extra_switches & 0x03); break;
+        case INP_SWG0:     value = global_extra_switches   & 0x04;  break;
+        case INP_SWG1:     value = !(global_extra_switches & 0x0c); break;
+        case INP_SWH0:     value = global_extra_switches   & 0x01;  break;
+        case INP_SWH1:     value = !(global_extra_switches & 0x03); break;
       }
     } else {
+      if ((~Transmitter.ignore_src & SWITCH_3x1) == SWITCH_3x1) {
```

Comment:
I don't like the large amount of highly repetitive code. Can you try to make it a loop over INP_SW* ?